clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchUtil.scala

```diff
@@ -21,7 +23,23 @@ object ElasticsearchUtil {
   class RequestException(val requestFailure: RequestFailure)
       extends IOException(s"Error in Elasticsearch request: ${requestFailure.error}")
 
-  def unpackResponse[T](response: Either[RequestFailure, RequestSuccess[T]]): T = {
-    response.fold(failure => throw new RequestException(failure), _.result)
+  /**
+    * Wrapper for elastic4's [[HttpClient]], used to add common patterns as extension methods.
+    */
+  implicit class HttpClientOps(val httpClient: HttpClient) extends AnyVal {
+
+    /**
+      * Send an HTTP request to Elasticsearch, and fold away the nested `Either`
+      * provided by elastic4s in the response.
+      */
+    def executeAndUnpack[T, U](request: T)(
```

Comment:
Thanks, this looks much better. But I now wonder if this name is helpful. Does the caller care that the response was "unpacked" before it was returned? Is this intermediate step important to the caller, or is it an implementation detail that the caller can ignore?

Reply:
Yeah, I struggled with the name too. I couldn't reuse `execute` because implicit wrappers can't overload existing methods... am open to suggestions if you have a better description than "unpack" for what this is doing.